python-package/lightgbm/callback.py

```diff
@@ -129,17 +129,41 @@ def record_evaluation(eval_result: Dict[str, Dict[str, List[Any]]]) -> Callable:
     if not isinstance(eval_result, dict):
         raise TypeError('eval_result should be a dictionary')
 
+    def _extract_data_and_eval_name(data_eval):
```

Comment:
I can't find where this function is used...

Reply:
I forgot to remove this in 914818c4bc70142c159d9865e0b1583486947c8d, removed in 54754fb83cd2826275a0b81a40b7d4742cb51dd5